services/wallet/transfer/transaction_manager_route.go

```diff
@@ -0,0 +1,346 @@
+	"github.com/status-im/status-go/services/wallet/router/pathprocessor"
+	"github.com/status-im/status-go/services/wallet/router/routes"
+	"github.com/status-im/status-go/transactions"
+)
+
+func (tm *TransactionManager) ClearLocalRouterTransactionsData() {
+	tm.routerTransactions = nil
+}
+
+func (tm *TransactionManager) ApprovalRequiredForPath(pathProcessorName string) bool {
+	for _, desc := range tm.routerTransactions {
+		if desc.routerPath.ProcessorName == pathProcessorName &&
+			desc.routerPath.ApprovalRequired {
+			return true
+		}
+	}
+	return false
+}
+
+func (tm *TransactionManager) ApprovalPlacedForPath(pathProcessorName string) bool {
+	for _, desc := range tm.routerTransactions {
+		if desc.routerPath.ProcessorName == pathProcessorName &&
+			desc.approvalTxSentHash != (types.Hash{}) {
+			return true
+		}
+	}
+	return false
+}
+
+func (tm *TransactionManager) TxPlacedForPath(pathProcessorName string) bool {
+	for _, desc := range tm.routerTransactions {
+		if desc.routerPath.ProcessorName == pathProcessorName &&
+			desc.txSentHash != (types.Hash{}) {
+			return true
+		}
+	}
+	return false
+}
+
+func (tm *TransactionManager) buildApprovalTxForPath(path *routes.Path, addressFrom common.Address,
+	usedNonces map[uint64]int64, signer ethTypes.Signer) (types.Hash, error) {
+	lastUsedNonce := int64(-1)
+	if nonce, ok := usedNonces[path.FromChain.ChainID]; ok {
+		lastUsedNonce = nonce
+	}
+
+	data, err := walletCommon.PackApprovalInputData(path.AmountIn.ToInt(), path.ApprovalContractAddress)
+	if err != nil {
+		return types.Hash{}, err
+	}
+
+	addrTo := types.Address(path.FromToken.Address)
+	approavalSendArgs := &transactions.SendTxArgs{
+		Version: transactions.SendTxArgsVersion1,
+
+		// tx fields
+		From:                 types.Address(addressFrom),
+		To:                   &addrTo,
+		Value:                (*hexutil.Big)(big.NewInt(0)),
+		Data:                 data,
+		Gas:                  (*hexutil.Uint64)(&path.ApprovalGasAmount),
+		MaxFeePerGas:         path.MaxFeesPerGas,
+		MaxPriorityFeePerGas: path.ApprovalPriorityFee,
+
+		// additional fields version 1
+		FromChainID: path.FromChain.ChainID,
+	}
+	if path.FromToken != nil {
+		approavalSendArgs.FromTokenID = path.FromToken.Symbol
+	}
+
+	builtApprovalTx, usedNonce, err := tm.transactor.ValidateAndBuildTransaction(approavalSendArgs.FromChainID, *approavalSendArgs, lastUsedNonce)
+	if err != nil {
+		return types.Hash{}, err
+	}
+	approvalTxHash := signer.Hash(builtApprovalTx)
+	usedNonces[path.FromChain.ChainID] = int64(usedNonce)
+
+	tm.routerTransactions = append(tm.routerTransactions, &RouterTransactionDetails{
+		routerPath:         path,
+		approvalTxArgs:     approavalSendArgs,
+		approvalTx:         builtApprovalTx,
+		approvalHashToSign: types.Hash(approvalTxHash),
+	})
+
+	return types.Hash(approvalTxHash), nil
+}
+
+func (tm *TransactionManager) buildTxForPath(path *routes.Path, addressFrom common.Address, addressTo common.Address,
```

Comment:
<img width="1020" alt="image" src="https://github.com/user-attachments/assets/adb6be5d-c034-418b-bbd2-46b321b03814">

This is indeed too many arguments. Can you please combine them in a single/few struct?

Reply:
Updated.